core/server/src/main/java/alluxio/master/file/FileSystemMaster.java

```diff
@@ -2205,18 +2207,15 @@ private void mountInternal(LockedInodePath inodePath, AlluxioURI ufsPath, boolea
       // Check that the ufsPath exists and is a directory
       UnderFileSystem ufs = UnderFileSystem.get(ufsPath.toString());
       ufs.setProperties(options.getProperties());
-      if (!ufs.exists(ufsPath.toString())) {
+      if (!ufs.isDirectory(ufsPath.toString())) {
         throw new IOException(
             ExceptionMessage.UFS_PATH_DOES_NOT_EXIST.getMessage(ufsPath.getPath()));
       }
-      if (ufs.isFile(ufsPath.toString())) {
-        throw new IOException(
-            ExceptionMessage.PATH_MUST_BE_DIRECTORY.getMessage(ufsPath.getPath()));
-      }
       // Check that the alluxioPath we're creating doesn't shadow a path in the default UFS
       String defaultUfsPath = Configuration.get(PropertyKey.UNDERFS_ADDRESS);
       UnderFileSystem defaultUfs = UnderFileSystem.get(defaultUfsPath);
-      if (defaultUfs.exists(PathUtils.concatPath(defaultUfsPath, alluxioPath.getPath()))) {
+      String shadowPath = PathUtils.concatPath(defaultUfsPath, alluxioPath.getPath());
+      if (defaultUfs.isFile(shadowPath) || defaultUfs.isDirectory(shadowPath)) {
```

Comment:
Do you think having `exists` is still useful? Its possible some under storages have a more optimized way of checking this.